### PR TITLE
Multiple error

### LINKED ISSUE #751: Diktat should not be printing the same error again when the Kotlin version is not specified
> Task :diktatFix
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version
[ERROR] 2021-02-07 23:51:16 Kotlin version not specified in the configuration file. Will be using 1.4.21 version

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
   Added atomic counter
   closes #751

### CODE REVIEW

File: diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt [hunk 1]
```diff
@@ -122,7 +123,9 @@ data class CommonConfiguration(private val configuration: Map<String, String>?)
      */
     val kotlinVersion: KotlinVersion by lazy {
         configuration?.get("kotlinVersion")?.kotlinVersion() ?: run {
```
Comment (petertrr) [inline diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt:125]: Did you figure out why it was possible? Body of lazy should have been executed only once, even with parallel execution. Counter fixes this behavior, of course, but it's always better to find a root cause. Did you try to reproduce it in maven, for example?

Comment (kentr0w) [reply diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt:125]: This was due to our `List<RulesConfig>.getCommonConfiguration()`  function. To fix, I can make an atomic counter (as already done) or make a lazy extension property.

Comment (petertrr) [reply diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt:125]: I suggest to use lazy extension property, I think using a function there was by mistake.

Comment (kentr0w) [reply diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt:125]: Unfortunately, a lazy property requires storing state, and there is no way to add state to an existing object from an extension property.

**Review by petertrr — APPROVED** [review]

**Pull request merged**